sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioDecoderCompat.java

```diff
@@ -0,0 +1,131 @@
+package com.smartdevicelink.api.audio;
+
+import android.content.Context;
+import android.media.MediaCodec;
+import android.media.MediaFormat;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+
+import com.smartdevicelink.api.audio.AudioStreamManager.SampleType;
+
+import java.lang.ref.WeakReference;
+import java.nio.ByteBuffer;
+
+/**
+ * The audio decoder to decode a single audio file to PCM.
+ * This decoder supports phones with api < 21 but uses methods deprecated with api 21.
+ */
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioDecoderCompat extends BaseAudioDecoder {
+    private static final String TAG = AudioDecoderCompat.class.getSimpleName();
+    private static final int DEQUEUE_TIMEOUT = 3000;
+
+    /**
+     * Creates a new object of AudioDecoder.
+     * @param audioSource The audio source to decode.
+     * @param context The context object to use to open the audio source.
+     * @param sampleRate The desired sample rate for decoded audio data.
+     * @param sampleType The desired sample type (8bit, 16bit, float).
+     * @param listener A listener who receives the decoded audio.
+     */
+    AudioDecoderCompat(Uri audioSource, Context context, int sampleRate, @SampleType int sampleType, AudioDecoderListener listener) {
```

Comment:
`@NonNull` should at least be added to `audioSource`, `context`

Reply:
Done